Tests/Unit/NetworkServer/ConcentratorDeduplicationTest.cs

```diff
@@ -156,26 +193,35 @@ public void When_Join_Request_Encountered_Should_Find_Duplicate(string station1,
         public void CreateKeyMethod_Should_Produce_Expected_Key_For_JoinRequests()
         {
             // arrange
-            var joinPayload = (LoRaPayloadJoinRequest)this.joinRequest.Payload;
-            joinPayload.DevNonce = new DevNonce(0);
-            this.joinRequest.SetPayload(joinPayload);
+            this.joinPayload.DevNonce = new DevNonce(0);
+            this.joinRequest.SetPayload(this.joinPayload);
 
             var expectedKey = "60-DA-A3-A5-F7-DB-FA-20-0F-8C-82-84-0E-CF-5B-42-64-0B-70-F3-B7-21-8A-4C-6B-BD-67-DB-54-2E-75-A4";
 
             // act/assert
-            Assert.Equal(expectedKey, ConcentratorDeduplication.CreateCacheKey(this.joinRequest));
+            Assert.Equal(expectedKey, this.concentratorDeduplication.CreateCacheKey(this.joinPayload));
         }
-        #endregion
 
         [Fact]
-        public void CreateKeyMethod_Should_Throw_When_Used_With_Wrong_Type()
+        public void CreateKeyMethod_Should_Include_All_Properties_For_JoinRequests()
         {
             // arrange
-            this.dataRequest.SetPayload(new Mock<LoRaPayloadJoinAccept>().Object);
+            var mock = new Mock<ConcentratorDeduplication>(MockBehavior.Default, this.cache, NullLogger<ConcentratorDeduplication>.Instance);
+            Memory<byte> actualBuffer = null;
+            _ = mock.Setup(x => x.HashKey(It.IsAny<byte[]>())).Callback<byte[]>(b => actualBuffer = b);
+            this.joinPayload.AppEUI = new byte[8] { 0, 1, 0, 1, 0, 1, 0, 1 };
+            this.joinPayload.DevEUI = new byte[8] { 1, 2, 3, 4, 5, 6, 7, 8 };
 
-            // act/assert
-            Assert.Throws<ArgumentException>(() => ConcentratorDeduplication.CreateCacheKey(this.dataRequest));
+            // act
+            _ = mock.Object.CreateCacheKey(this.joinPayload);
+
+            // assert
+            mock.Verify(x => x.HashKey(It.IsAny<byte[]>()), Times.Once);
+            Assert.True(MemoryExtensions.SequenceEqual(this.joinPayload.AppEUI.Span, actualBuffer.Span[0..8]));
+            Assert.True(MemoryExtensions.SequenceEqual(this.joinPayload.DevEUI.Span, actualBuffer.Span[8..16]));
```

Comment:
Same as [earlier comment](url):

```suggestion
            Assert.True(this.joinPayload.AppEUI.Span.SequenceEqual(actualBuffer.Span[0..8]));
            Assert.True(this.joinPayload.DevEUI.Span.SequenceEqual(actualBuffer.Span[8..16]));
```

Reply:
See comment above, code was removed.